fineract-provider/src/main/java/org/apache/fineract/infrastructure/event/external/service/message/domain/MessageBusinessDate.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.infrastructure.event.external.service.message.domain;
+
+import java.time.LocalDate;
+import java.util.Objects;
+import lombok.Getter;
+
+@Getter
+public class MessageBusinessDate {
+
+    private LocalDate businessDate;
+
+    public MessageBusinessDate(LocalDate businessDate) {
+        this.businessDate = Objects.requireNonNull(businessDate, "businessDate cannot be null");
```

Comment:
Please extract the string into a constant.

Reply:
Hmm, I can do it but I'm not sure what value we add here (and I know I ask everybody to extract magic strings to constants). 
But here, this is not a magic string. There's no logic building upon this, only an exception's message.
What do you think Adam?

Reply:
My concern is every time the Objects, requireNonNull method got invoked (and it will) this string will be saved into memory as a "new" string object and not reused

Reply:
Resolved :)